torch/testing/_internal/common_utils.py

```diff
@@ -1322,13 +1300,12 @@ def assertWarnsOnceRegex(self, category, regex=''):
                 yield
             finally:
                 torch.set_warn_always(prev)
-                if len(ws) == 0:
-                    self.fail('no warning caught')
-                if len(ws) > 1:
-                    self.fail('too many warnings caught: %s' % '\n    '.join([str(w) for w in ws]))
-                self.assertTrue(type(ws[0].message) is category)
-                self.assertTrue(re.match(pattern, str(ws[0].message)),
-                                f'{pattern}, {ws[0].message}')
+            if len(ws) == 0:
+                self.fail('no warning caught')
+            for w in ws:
+                self.assertTrue(type(w.message) is category)
+                self.assertTrue(re.match(pattern, str(w.message)),
+                                f'{pattern}, {w.message}')
```

Comment:
Unindent this stanza since it should not be part of `finally`: if an exception is raised (for instance `SkipTest`) the warning may not have been emitted.